packages/data/src/components/with-select/index.js

```diff
@@ -53,7 +54,7 @@ import useSelect from '../use-select';
 const withSelect = ( mapSelectToProps ) =>
 	createHigherOrderComponent(
 		( WrappedComponent ) =>
-			pure( ( ownProps ) => {
+			memo( ( ownProps ) => {
```

Comment:
@Mamaduka After testing the mobile native version, I noticed a nuance here. When using `memo`, if I try to get the reference via the `ref` prop in the resulting component, I get the following warning:

```
Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?
```

I explored further and realized that the difference between using `pure` and `memo` is that each returns different types:
- [`pure`](https://github.com/WordPress/gutenberg/blob/63ef497cfca9c9e2214334d885736efe4db3a043/packages/compose/src/higher-order/pure/index.tsx#L32-L51) returns class component.
- `memo` returns function component.

For the latter, I can't use `forwardRef` anymore for forwarding the ref after using `withSelect`. I wonder if we should implement a mechanism for this. WDYT?

Reply:
Thanks for looking into this, @fluiddot!

Can you link me to one of these components? I would like to experiment a bit with ideas.

Reply:
We should deprecate withSelect too :P